consensus/dummy/dynamic_fees.go

```diff
@@ -37,22 +37,35 @@ var (
 
 // CalcExtraPrefix takes the previous header and the timestamp of its child
 // block and calculates the expected extra prefix for the child block.
-//
-// CalcExtraPrefix should only be called if [timestamp] >= [config.ApricotPhase3Timestamp]
 func CalcExtraPrefix(
 	config *params.ChainConfig,
 	parent *types.Header,
 	timestamp uint64,
 ) ([]byte, error) {
-	window, err := calcFeeWindow(config, parent, timestamp)
-	return window.Bytes(), err
+	switch {
+	case config.IsApricotPhase3(timestamp):
+		window, err := calcFeeWindow(config, parent, timestamp)
+		return window.Bytes(), err
+	default:
+		// Prior to AP3 there was no expected extra prefix.
+		return nil, nil
+	}
```

Comment:
This switch on `timestamp` will need to exist with ACP-176, the question is just if the default case should error (this function shouldn't be called before AP3) or if the function handles it as it does here.

Same comment on `CalcBaseFee` below.